### PR TITLE
Features/42-read-asc-andor

### PR DESCRIPTION
**Pull request by sangttruong**
Add unit test for read_asc_Andor.R

### CODE REVIEW

**Comment by GegznaV** [comment]
Please, always add references to linked/related issues.

Related #42

File: R/read_asc_Andor.R [hunk 1]
```diff
@@ -49,11 +49,33 @@ read_asc_Andor <- function(file = stop("filename or connection needed"),
...
-    skip("Need to adapt to new package")
-    expect_known_hash(
-      read_asc_Andor("fileio/asc.Andor/ASCII-Andor-Solis.asc"),
-      "9ead937f51"
-    )
+
+  andor <- system.file("extdata/fileio/asc.Andor/", "ASCII-Andor-Solis.asc",package = "hySpc.read.txt")
+  spc <- read_asc_Andor(andor)
+
+  test_that("Andor Solis .asc filename",{
```
Comment (GegznaV) [inline R/read_asc_Andor.R]: Maybe better not "Andor Solis .asc filename" but "Andor Solis .asc: labels are correct"

File: R/read_asc_Andor.R [hunk 2]
```diff
@@ -49,11 +49,33 @@ read_asc_Andor <- function(file = stop("filename or connection needed"),
...
+                 c("161.408","165.729","170.046","174.361","178.672","182.981","187.285","191.588","195.887",
+                   "200.184","204.477","208.767","213.054","217.339","221.62", "225.898","230.173","234.445",
+                   "238.714","242.981","247.244","251.505","255.762","260.016","264.268","268.516","272.762",
+                   "277.004","281.244","285.48", "289.714","293.945","298.172","302.397","306.619","310.838",
+                   "315.054","319.267","323.477","327.685","331.889","336.091","340.289","344.484","348.677",
+                   "352.867","357.054","361.238","365.419","369.596","373.772","377.945","382.114","386.281",
+                   "390.444","394.605","398.763","402.919","407.071","411.22", "415.366","419.51", "423.651"))
+  })
+
+  test_that("Andor Solis .asc wavelength", {
```
Comment (GegznaV) [inline R/read_asc_Andor.R]: Change to "Andor Solis .asc: wavelengths are correct"

File: R/read_asc_Andor.R [hunk 3]
```diff
@@ -49,11 +49,33 @@ read_asc_Andor <- function(file = stop("filename or connection needed"),
...
 hySpc.testthat::test(read_asc_Andor) <- function() {
   context("read_asc_Andor")
-  test_that("Andor Solis .asc text files", {
-    skip("Need to adapt to new package")
-    expect_known_hash(
-      read_asc_Andor("fileio/asc.Andor/ASCII-Andor-Solis.asc"),
-      "9ead937f51"
-    )
+
+  andor <- system.file("extdata/fileio/asc.Andor/", "ASCII-Andor-Solis.asc",package = "hySpc.read.txt")
```
Comment (GegznaV) [inline R/read_asc_Andor.R]: Add space after comma

File: R/read_asc_Andor.R [hunk 4]
```diff
@@ -49,11 +49,33 @@ read_asc_Andor <- function(file = stop("filename or connection needed"),
...
+  test_that("Andor Solis .asc labels are correct",{
+    expect_equal(spc@label$.wavelength, NULL)
+    expect_equal(spc@label$spc, NULL)
+    expect_equal(spc@label$filename, "filename")
+  })
+
+  test_that("Andor Solis .asc spectral data", {
+    expect_equal(dim(spc@data$spc), c(5,63))
+
+    expect_equal(colnames(spc@data$spc),
```
Comment (GegznaV) [inline R/read_asc_Andor.R]: I asked Sang to check if wavelengths are read correctly. Just I'm wondering if we should check all the values or just a few (smallest, largest, and several intermediate). What do you think @bryanhanson, @cbeleites?

This concerns this one and especially the future unit tests.

File: R/read_dat_Witec.R [hunk 5]
```diff
@@ -58,7 +58,7 @@ hySpc.testthat::test(read_dat_Witec) <- function() {
   test_that("-y file guessing", {
     spc <- read_dat_Witec(paste0(tmpdir, "/Witec-timeseries-x.dat"))
     spc$filename <- gsub("^.*/", "", spc$filename)
-
+    capture_output(paste0(tmpdir, "/Witec-timeseries-x.dat"))
```
Comment (GegznaV) [inline R/read_dat_Witec.R]: Why do we need to capture the output of `paste0()`? Why is the captured output not used?

File: R/read_asc_Andor.R [hunk 6]
```diff
@@ -49,11 +49,33 @@ read_asc_Andor <- function(file = stop("filename or connection needed"),
...
+  andor <- system.file("extdata/fileio/asc.Andor/", "ASCII-Andor-Solis.asc",package = "hySpc.read.txt")
+  spc <- read_asc_Andor(andor)
+
+  test_that("Andor Solis .asc labels are correct",{
+    expect_equal(spc@label$.wavelength, NULL)
+    expect_equal(spc@label$spc, NULL)
+    expect_equal(spc@label$filename, "filename")
+  })
+
+  test_that("Andor Solis .asc spectral data", {
```
Comment (GegznaV) [inline R/read_asc_Andor.R]: Change to "Andor Solis .asc: spectra are correct"

**Review by GegznaV — CHANGES_REQUESTED** [review]

**Comment by bryanhanson** [comment]
If you are going to check at all you may as well compare the whole vector. But for my money the check on the dimensions would be enough.

File: R/read_asc_Andor.R [hunk 7]
```diff
@@ -63,4 +63,33 @@ read_asc_Andor <- function(file = stop("filename or connection needed"),
...
+                   "277.004","281.244","285.48", "289.714","293.945","298.172","302.397","306.619","310.838",
+                   "315.054","319.267","323.477","327.685","331.889","336.091","340.289","344.484","348.677",
+                   "352.867","357.054","361.238","365.419","369.596","373.772","377.945","382.114","386.281",
+                   "390.444","394.605","398.763","402.919","407.071","411.22", "415.366","419.51", "423.651"))
+  })
+
+  test_that("Andor Solis .asc wavelength", {
+    expect_equal(length(spc@wavelength), 63)
+
+    expect_equal(round(spc@wavelength[[47]], 3), 357.054)
```
Comment (cbeleites) [inline R/read_asc_Andor.R:92]: `@wavelength` is a vector: use single square brackets

Comment (cbeleites) [inline R/read_asc_Andor.R]: `@wavelength` is a vector: use single square brackets

File: R/read_asc_Andor.R (hunk 4 above)
Comment (cbeleites) [reply R/read_asc_Andor.R]: Checking the vector isn't any more work (using `dput()` on manually checked values)

File: R/read_asc_Andor.R [hunk 8]
```diff
@@ -47,9 +47,9 @@ read_asc_Andor <- function(file = stop("filename or connection needed"),
 
 # Unit tests -----------------------------------------------------------------
 
-# hySpc.testthat::test(read_asc_Andor) <- function() {
-#   context("read_asc_Andor")
-#
+hySpc.testthat::test(read_asc_Andor) <- function() {
+  context("read_asc_Andor")
+
 #   tmpdir <- paste0(tempdir(), "/test_Ancor")
```
Comment (cbeleites) [inline R/read_asc_Andor.R]: Why is there commented code left in here?

**Review by cbeleites — CHANGES_REQUESTED** [review]

**Comment by GegznaV** [comment]
Related #42

File: R/read_asc_Andor.R (hunk 4 above)
Comment (GegznaV) [reply R/read_asc_Andor.R]: But I think that it would make unit tests less readable if the output of `dput()` would be added. Especially for large files.

**Review by GegznaV — APPROVED** [review]

**Comment by GegznaV** [comment]
@cbeleites, I'm merging this now. If you'd like additional improvements, please, open a new issue for that. OK?

**Pull request merged**